### PR TITLE
Add wallet find command

### PR DESCRIPTION
**Pull request by gracegrape**

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/wallet/WalletFindCommand.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import seedu.address.model.transaction.DescriptionContainsKeywordsPredicate;
+
+/**
+ * Finds and lists all transactions in wallet book whose name contains any of the argument keywords.
+ * Keyword matching is case insensitive.
+ */
+public class WalletFindCommand extends Command {
+    public static final String COMMAND_WORD = "find";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all transactions whose names contain any of "
```
Comment (CheyanneSim) [inline src/main/java/seedu/address/logic/commands/wallet/WalletFindCommand.java]: I think shouldn't use the word 'whose' since transaction is not person, but at the same time i also don't really know how to phrase it haha. which? with?

File: src/test/java/seedu/address/commons/util/StringUtilTest.java [hunk 2]
```diff
@@ -109,7 +109,7 @@ public void containsWordIgnoreCase_validInputs_correctResult() {
         assertFalse(StringUtil.containsWordIgnoreCase("    ", "123"));
 
         // Matches a partial word only
-        assertFalse(StringUtil.containsWordIgnoreCase("aaa bbb ccc", "bb")); // Sentence word bigger than query word
+        assertTrue(StringUtil.containsWordIgnoreCase("aaa bbb ccc", "bb")); // Sentence word bigger than query word
```
Comment (CheyanneSim) [inline src/test/java/seedu/address/commons/util/StringUtilTest.java:112]: [Question] why did you change this to true though? I thought this is for the method containsWordIgnoreCase and not equalsSubWordIgnoreCase. Did this pass the test? O.o 

Also, did you add test for equalsSubWordIgnoreCase?

File: src/test/java/seedu/address/logic/commands/wallet/WalletFindCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        WalletFindCommand findFirstCommandCopy = new WalletFindCommand(firstPredicate);
+        assertTrue(findFirstCommand.equals(findFirstCommandCopy));
+
+        // different types -> returns false
+        assertFalse(findFirstCommand.equals(1));
+
+        // null -> returns false
+        assertFalse(findFirstCommand.equals(null));
+
+        // different transaction -> returns false
```
Comment (CheyanneSim) [inline src/test/java/seedu/address/logic/commands/wallet/WalletFindCommandTest.java]: different predicate* haha

File: src/test/java/seedu/address/logic/commands/wallet/WalletFindCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+public class WalletFindCommandTest {
+    private Model model = new ModelManager(getTypicalAddressBook(), getTypicalWallet(), new UserPrefs());
+    private Model expectedModel = new ModelManager(getTypicalAddressBook(), getTypicalWallet(), new UserPrefs());
+
+    @Test
+    public void equals() {
+        DescriptionContainsKeywordsPredicate firstPredicate =
+                new DescriptionContainsKeywordsPredicate(Collections.singletonList("first"));
+        DescriptionContainsKeywordsPredicate secondPredicate =
```
Comment (CheyanneSim) [inline src/test/java/seedu/address/logic/commands/wallet/WalletFindCommandTest.java:32]: [Question] Did you change the test method in DescriptionContainsKeywordPredicate to use StringUtil.equalsSubWordIgnoreCase?

**Review by CheyanneSim — COMMENTED** [review]
Thank you for the great work!! :)

File: src/main/java/seedu/address/commons/util/StringUtil.java [hunk 5]
```diff
@@ -36,7 +36,22 @@ public static boolean containsWordIgnoreCase(String sentence, String word) {
         String[] wordsInPreppedSentence = preppedSentence.split("\\s+");
```
Comment (joel-lim) [inline src/main/java/seedu/address/commons/util/StringUtil.java:35]: If we're allowing partial matches probably don't need to split the sentence up anymore right?
Like just return sentence.contains(preppedWord) (after toLowerCase-ing them)

File: src/test/java/seedu/address/commons/util/StringUtilTest.java [hunk 6]
```diff
@@ -109,7 +109,7 @@ public void containsWordIgnoreCase_validInputs_correctResult() {
         assertFalse(StringUtil.containsWordIgnoreCase("    ", "123"));
```
Comment (joel-lim) [inline src/test/java/seedu/address/commons/util/StringUtilTest.java:110]: Should adjust the test docstring above also to reflect that it checks for partial word matches

File: src/test/java/seedu/address/logic/commands/wallet/WalletFindCommandTest.java [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+        String expectedMessage = String.format(MESSAGE_TRANSACTIONS_LISTED_OVERVIEW, 3);
+        DescriptionContainsKeywordsPredicate predicate = preparePredicate("TA MRT RICE");
+        WalletFindCommand command = new WalletFindCommand(predicate);
+        expectedModel.updateFilteredTransactionList(predicate);
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(TA_JOB, DUCK_RICE, MRT_CONCESSION), model.getFilteredTransactionList());
+    }
+
+    /**
+     * Parses {@code userInput} into a {@code NameContainsKeywordsPredicate}.
```
Comment (joel-lim) [inline src/test/java/seedu/address/logic/commands/wallet/WalletFindCommandTest.java]: Change to DescriptionContainsKeywordsPredicate

File: src/main/java/seedu/address/logic/commands/wallet/WalletFindCommand.java (hunk 1 above)
Comment (joel-lim) [reply src/main/java/seedu/address/logic/commands/wallet/WalletFindCommand.java]: wah cheyanne grammar nazi
I think whose is fine HAHA

Comment (gracegrape) [reply src/main/java/seedu/address/logic/commands/wallet/WalletFindCommand.java]: how about 
`Finds all transactions with description containing any of... `
HAHA

File: src/test/java/seedu/address/commons/util/StringUtilTest.java (hunk 2 above)
Comment (gracegrape) [reply src/test/java/seedu/address/commons/util/StringUtilTest.java:112]: Regarding the first part, because originally, the containsWordIgnoreCase only works for the full word. Then I changed it to work for subword, so for "Rice", "ri" works as well, thus no longer assertFalse, since "aaa bbb ccc" contains "bb".

For the 2nd part, thanks! Will add it in.

File: src/test/java/seedu/address/logic/commands/wallet/WalletFindCommandTest.java (hunk 3 above)
Comment (gracegrape) [reply src/test/java/seedu/address/logic/commands/wallet/WalletFindCommandTest.java]: hahaha thanks!!

File: src/test/java/seedu/address/logic/commands/wallet/WalletFindCommandTest.java (hunk 4 above)
Comment (gracegrape) [reply src/test/java/seedu/address/logic/commands/wallet/WalletFindCommandTest.java:32]: Thank you! Just added it.

File: src/main/java/seedu/address/commons/util/StringUtil.java (hunk 5 above)
Comment (gracegrape) [reply src/main/java/seedu/address/commons/util/StringUtil.java:35]: Oh yea! True, didn't consider this part. Thank you! Will edit my code.

File: src/test/java/seedu/address/commons/util/StringUtilTest.java (hunk 6 above)
Comment (gracegrape) [reply src/test/java/seedu/address/commons/util/StringUtilTest.java:110]: Alright! Thanks, will do,
